File: kaop/src/main/java/io/github/coyamo/kaop/ProxyJoinPoint.java
package io.github.coyamo.kaop;

import androidx.annotation.Nullable;

import java.lang.reflect.Method;

class ProxyJoinPoint extends AbsJoinPoint{
    @Nullable private final Object result;
    private final AbsJoinPoint joinPoint;


    public ProxyJoinPoint(AbsJoinPoint joinPoint,@Nullable Object result){
        this.joinPoint = joinPoint;
        this.result = result;
    }
    @Override
    public Object proceed() {
        return result;
    }

    @Override
    public Object getOwner() {
        return joinPoint.getOwner();
    }

    @Override
    public Method getMethod() {
        return joinPoint.getMethod();
    }

    @Override
    public Object[] getParams() {
        return joinPoint.getParams();
    }
}

File: kaop/src/main/java/io/github/coyamo/kaop/ProxyJoinPoint.kt
package io.github.coyamo.kaop

import java.lang.reflect.Method

/**
 * @author Coyamo
 * @date 2022/12/17 00:47
 * @version 1.0
 */
internal class ProxyJoinPoint(private val joinPoint: AbsJoinPoint, private val result: Any?) :
    AbsJoinPoint() {
    override fun proceed(): Any? {
        return result
    }

    override val owner: Any
        get() = joinPoint.owner
    override val method: Method
        get() = joinPoint.method
    override val params: Array<Any?>
        get() = joinPoint.params
}